### PR TITLE
add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rodrigomelo9**

### CODE REVIEW

File: ADD_NEW_DISTRO.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# How to add a new distro
+
+## Add Dockerfiles
+
+Create `build_<DISTRO>.dockerfile` and `run_<DISTRO>.dockerfile` in the root directory.
+
+* These are based on [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).
+* Only install the needed dependencies before the GHDL compilation/installation.
```
Comment (eine) [inline ADD_NEW_DISTRO.md:8]: "Images should contain the dependencies for building or installing GHDL only. Do not include the actual building/installation procedures in the dockerfiles".

File: ADD_NEW_DISTRO.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# How to add a new distro
+
+## Add Dockerfiles
+
+Create `build_<DISTRO>.dockerfile` and `run_<DISTRO>.dockerfile` in the root directory.
+
+* These are based on [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).
```
Comment (eine) [inline ADD_NEW_DISTRO.md:7]: "Add one stage for each (supported) backend: `mcode`, `llvm` and/or `gcc`".

File: ADD_NEW_DISTRO.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+* Only install the needed dependencies before the GHDL compilation/installation.
+
+## CI
+
+* Add a new `case` inside of `create ()` (`run.sh`)
```
Comment (eine) [inline ADD_NEW_DISTRO.md:12]: "In `run.sh`, add a new..."

**Review by eine — COMMENTED** [review]
Overall, looks good. However, let's wait until we are done with adding centos, beause additional steps/hints might show.

Maybe we should rename this file to `CONTRIBUTING.md`?

**Comment by rodrigomelo9** [comment]
> However, let's wait until we are done with adding centos, beause additional steps/hints might show.

Of course, because that it is a draft ;-)

> Maybe we should rename this file to CONTRIBUTING.md?

Done. Let me know if ok in the root (I saw that is under `.github` in `hdl/awesome`, which is not common in general).

**Comment by eine** [comment]
> > Maybe we should rename this file to CONTRIBUTING.md?
> 
> Done. Let me know if ok in the root (I saw that is under `.github` in `hdl/awesome`, which is not common in general).

Having it in `.github/CONTRIBUTING.md` is one of the two standard locations for GitHub to show it in https://github.com/ghdl/docker/community. See https://docs.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors. In this case, I'm good with having it in the root.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+* Add the desidered `<DISTRO>` and its `<VERSIONS>` to the matrix of the `base.yml` CI workflow:
+```yaml
+matrix:
+  include: [
+    { distro: arch,     version: ''        },
+    { distro: debian,   version: buster    },
+    ...
+    { distro: <DISTRO>, version: <VERSION> },
+```
+* Add the desidered `<DISTRO><VERSION>` and `<BACKEND>` to the matrix of the `test.yml` CI workflow:
```
Comment (eine) [inline CONTRIBUTING.md:28]: Maybe we should clarify that the version of the backend needs to be specified for GCC and for LLVM on Debian/Ubuntu: https://github.com/ghdl/docker/blob/master/.github/workflows/test.yml#L20-L37

Comment (rodrigomelo9) [reply CONTRIBUTING.md:28]: Sorry, I have not clear that point. I don't know why is specified with Debian and Ubuntu, and not in Fedora, and in the case of centos, is not the version of the backend, is the version of the OS (something to provide a different stage name).

Comment (eine) [reply CONTRIBUTING.md:28]: On Fedora, the argument to `configure` is `--with-llvm-config=llvm-config`. However, on Debian/Ubuntu executable `llvm-config` does not exist. It is named `llvm-config-VERSION`. That's why we need to specify it.

On the other hand, on CentOS, the version of the distribution is provided in field `distro`. Field `backend` contains the name and version of the backend. Hence `distro: centos7,  backend: gcc-7` is read as "build GCC v7 on CentOS 7". The same way `distro: bullseye, backend: gcc-9.1.0` means "build GCC v9.1.0 on Debian Bullseye". It's unfortunate that you chose to use GCC v7 on CentOS 7 and GCC v8 on CentOS 8. That's misleading.

>  is the version of the OS (something to provide a different stage name)

This is the case in workflow `base.yml`, NOT in `test.yml` (which we are talking about here).

Comment (rodrigomelo9) [reply CONTRIBUTING.md:35]: Hi. I tried to clarify this and other points. Let me know what do you think.

**Review by eine — APPROVED** [review]

**Comment by eine** [comment]
Thanks!

**Pull request merged**